transaction_test.go

```diff
@@ -1902,3 +1903,120 @@ func TestRollbackTransaction(t *testing.T) {
 	IncrementHelper(t, b, counterAddress, addTwoScript, 2)
 
 }
+
+func Test_SimulatedMainnetTransaction(t *testing.T) {
```

Comment:
Are these tests actually connecting to the real archive nodes? It might make sense to mock/stub this away, to avoid depending on such a remote system and avoid flakiness when there are connection or availability issues

Reply:
Already experienced some flakiness so yeah, good idea for sure.

Reply:
But will have to look into how hard it is to mock all the requests, as there are so many I believe.

Reply:
> Are these tests actually connecting to the real archive nodes? It might make sense to mock/stub this away, to avoid depending on such a remote system and avoid flakiness when there are connection or availability issues

@turbolent I mocked the Archive client and used it to test but it's proving very hard to write a good test since there are a lot of register values being requested and transferred. I'm not sure is there's a better way, so let me know if you have an idea. 
Otherwise I excluded this test from transaction tests and moved them to separate test file with `integration` tag.

Reply:
we could run the test against mainnet and dump all the registers read into a file, then switch the tests for read from that file.

I guess the file would need to be updated from time to time, but as long as we have some instructions on how to do it it should be fine right?

Reply:
This could work, let me take a stab at it.